### PR TITLE
Friending

### PR DESCRIPTION
**Pull request by CaptainStack**
* Updates to views and styling to incorporate new feature
* Migration to add new friendship table to database
* Add friendship model
* Put User access to friendship and friends in Friendship_Helper concern
* Move omniauth functions from user model to Omniauth_Helper concern
* Add User profiles
* Add routes for friendship management

### CODE REVIEW

**Comment by ashishsc** [comment]
The commit message is kind of bad 'Finalize friending' something more descriptive should be used or a squashing

File: app/models/user.rb [hunk 1]
```diff
@@ -8,6 +8,15 @@ class User < ActiveRecord::Base
 
   has_many :user_movies
   
+  def friends
+    # Update when migrated to Rails 5 to use '.or.where()'
+    Friendship.where("user_id = ? or friend_id = ?", self.id, self.id)
+  end
+  
+  def pending_friend_requests
+    self.friends.select { |friend| friend.is_pending }
```
Comment (ashishsc) [inline app/models/user.rb]: im confused about what this line does.... the select return a subset that satisfies the blocks' conditions? similar to 
```javascript
Array.filter()
```
?

File: app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    redirect_to(:back)
+  end
+  
+  private
+    def friendship_params
+      params[:friendship] = params;
+      params.require(:friendship).permit(:user_id, :friend_id, :is_pending)
+    end
+    
+    def correct_user
```
Comment (ashishsc) [inline app/controllers/friendships_controller.rb]: can i get a comment on what this does

File: app/controllers/friendships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  private
+    def friendship_params
+      params[:user_id] = current_user.id
+      params[:friendship] = params;
+      params.require(:friendship).permit(:user_id, :friend_id, :is_pending)
+    end
+
+    def can_delete
```
Comment (ashishsc) [inline app/controllers/friendships_controller.rb]: all these boolean fns should follow the idiomatic ruby convention of ?

File: app/controllers/friendships_controller.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      @friendship = current_user.find_friendship_by_friend_id(params[:friend_id])
+      redirect_to root_url if @friendship.nil?
+    end
+
+    def can_accept
```
Comment (ashishsc) [inline app/controllers/friendships_controller.rb]: all these boolean fns should follow the idiomatic ruby convention of ?
 Add a line note

File: db/migrate/20160515205444_create_friendships.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreateFriendships < ActiveRecord::Migration
+  def change
+    create_table :friendships do |t|
+      t.integer :user_id
```
Comment (ashishsc) [inline db/migrate/20160515205444_create_friendships.rb]: Do we need a primary key?

File: app/controllers/friendships_controller.rb (hunk 4 above)
Comment (CaptainStack) [inline app/controllers/friendships_controller.rb]: I think no for a few reasons. They don't necessarily return a boolean. In some cases they will redirect. They also are not really generic functions that are intended to be used outside of their controller. They are only intended to be run as specified at the top of the controller file.

File: app/models/user.rb (hunk 1 above)
Comment (CaptainStack) [inline app/models/user.rb]: I think select is just what filter is called in Ruby.

File: app/controllers/friendships_controller.rb (hunk 2 above)
Comment (CaptainStack) [inline app/controllers/friendships_controller.rb]: Not going with that anymore.

**Pull request merged**